app/src/main/java/com/example/mona/facebookoffline/EditActivity.java

```diff
@@ -20,12 +26,22 @@
 
     private EditText mPostBody;
 
+    // Key used for saving mPhotoUris across different instances
+    private static final String PHOTO_URI_LIST = "uris";
+    private ArrayList<Uri> mPhotoUris;
+
     @Override
     protected void onCreate(Bundle savedInstanceState) {
         super.onCreate(savedInstanceState);
         ((FacebookOfflineApp) getApplication()).getAppComponent().inject(this);
         setContentView(R.layout.activity_edit);
 
+        if (savedInstanceState != null) {
+            mPhotoUris = (ArrayList) savedInstanceState.getSerializable(PHOTO_URI_LIST);
```

Comment:
Good catch! I think so if we somehow save the instance state without setting `PHOTO_URI_LIST`. Will fix, thanks